gwctl/pkg/printer/httproutes.go

```diff
@@ -81,50 +78,125 @@ func (hp *HTTPRoutesPrinter) PrintTable(resourceModel *resourcediscovery.Resourc
-	EffectivePolicies        any                         `json:",omitempty"`
-}
-
 func (hp *HTTPRoutesPrinter) PrintDescribeView(resourceModel *resourcediscovery.ResourceModel) {
 	index := 0
+
 	for _, httpRouteNode := range resourceModel.HTTPRoutes {
 		index++
 
-		views := []httpRouteDescribeView{
-			{
-				Name:      httpRouteNode.HTTPRoute.GetName(),
-				Namespace: httpRouteNode.HTTPRoute.GetNamespace(),
-			},
-			{
-				Hostnames:  httpRouteNode.HTTPRoute.Spec.Hostnames,
-				ParentRefs: httpRouteNode.HTTPRoute.Spec.ParentRefs,
-			},
+		metadata := httpRouteNode.HTTPRoute.ObjectMeta.DeepCopy()
+		resetMetadataFields(metadata)
+
+		namespace := handleDefaultNamespace(httpRouteNode.HTTPRoute.Namespace)
+
+		pairs := []*DescriberKV{
+			{"Name", httpRouteNode.HTTPRoute.GetName()},
+			{"Namespace", namespace},
+			{"Label", httpRouteNode.HTTPRoute.Labels},
+			{"Annotations", httpRouteNode.HTTPRoute.Annotations},
+			{"APIVersion", httpRouteNode.HTTPRoute.APIVersion},
+			{"Kind", httpRouteNode.HTTPRoute.Kind},
+			{"Metadata", metadata},
+			{"Spec", httpRouteNode.HTTPRoute.Spec},
+			{"Status", httpRouteNode.HTTPRoute.Status},
 		}
-		if policyRefs := resourcediscovery.ConvertPoliciesMapToPolicyRefs(httpRouteNode.Policies); len(policyRefs) != 0 {
-			views = append(views, httpRouteDescribeView{
-				DirectlyAttachedPolicies: policyRefs,
-			})
+
+		// DirectlyAttachedPolicies
+		directlyAttachedPolicies := &Table{
+			ColumnNames:  []string{"Type", "Name"},
+			UseSeparator: true,
 		}
-		if len(httpRouteNode.EffectivePolicies) != 0 {
-			views = append(views, httpRouteDescribeView{
-				EffectivePolicies: httpRouteNode.EffectivePolicies,
-			})
+
+		for _, policyNode := range httpRouteNode.Policies {
+			if policyNode.Policy.IsDirect() {
+				policyNamespace := handleDefaultNamespace(policyNode.Policy.TargetRef().Namespace)
+
+				row := []string{
+					// Type
+					fmt.Sprintf("%v.%v", policyNode.Policy.Unstructured().GroupVersionKind().Kind, policyNode.Policy.Unstructured().GroupVersionKind().Group),
+					// Name
+					fmt.Sprintf("%v/%v", policyNamespace, policyNode.Policy.Unstructured().GetName()),
+				}
+
+				directlyAttachedPolicies.Rows = append(directlyAttachedPolicies.Rows, row)
+			}
+		}
+
+		if len(directlyAttachedPolicies.Rows) != 0 {
+			pairs = append(pairs, &DescriberKV{Key: "DirectlyAttachedPolicies", Value: directlyAttachedPolicies})
 		}
 
-		for _, view := range views {
-			b, err := yaml.Marshal(view)
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "failed to marshal to yaml: %v\n", err)
-				os.Exit(1)
+		// InheritedPolicies
+		if len(httpRouteNode.InheritedPolicies) != 0 {
+			inheritedPolicies := &Table{
+				ColumnNames:  []string{"Type", "Name", "Target Kind", "Target Name"},
+				UseSeparator: true,
 			}
-			fmt.Fprint(hp, string(b))
+
+			for _, policyNode := range httpRouteNode.InheritedPolicies {
+				policyNamespace := handleDefaultNamespace(policyNode.Policy.Unstructured().GetNamespace())
```

Comment:
Two things:
- I don't think you need `handleDefaultNamespace` to normalize. It's only useful in unit tests and I'd rather prefer us adding the namespace to the resource manually when creating it. Resources in real context should always have namespace
- This is also error prone for cases where the policy is a cluster-scoped resource (so it will not have a namespace and "default" is incorrect in such a case). You may need to handle that